test/SlowTests/Voron/CompactTrees.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using FastTests.Voron;
+using Tests.Infrastructure;
+using Voron.Data.CompactTrees;
+using Xunit;
+using Xunit.Abstractions;
+using static Voron.Data.CompactTrees.CompactTree;
+
+namespace SlowTests.Voron
+{
+
+    public class CompactTreesTest : StorageTest
+    {
+        public CompactTreesTest(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        public enum SamplingMethod
+        {
+            FullRandom,
+            BranchRandom,
+            FullScan,
+        }
+
+        [RavenTheory(RavenTestCategory.Voron)]
+        [InlineData(1337, SamplingMethod.FullScan)]
+        [InlineData(1337, SamplingMethod.FullRandom)]
+        [InlineData(1337, SamplingMethod.BranchRandom)]
+        public void CanRecompressItemsWithDeletesAndInserts(int seed, SamplingMethod samplingMode)
+        {
+            static void Shuffle(string[] list, Random rng)
+            {
+                int n = list.Length;
+                while (n > 1)
+                {
+                    n--;
+                    int k = rng.Next(n + 1);
+                    var value = list[k];
+                    list[k] = list[n];
+                    list[n] = value;
+                }
+            }
+
+            const int Size = 200000;
```

Comment:
Shouldn't we make it as part of `InlineData` with multiple different values instead of sticking to this single hard coded size?